File: src/main/java/com/starshootercity/abilities/MoreExhaustion.java
package com.starshootercity.abilities;

import com.starshootercity.OriginSwapper;
import net.kyori.adventure.key.Key;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.entity.EntityExhaustionEvent;
import org.jetbrains.annotations.NotNull;

import java.util.List;

public class MoreExhaustion implements VisibleAbility, Listener {
    @Override
    public @NotNull Key getKey() {
        return Key.key("origins:more_exhaustion");
    }

    @Override
    public @NotNull List<OriginSwapper.LineData.LineComponent> getDescription() {
        return OriginSwapper.LineData.makeLineFor("You exhaust much quicker than others, thus requiring you to eat more.", OriginSwapper.LineData.LineComponent.LineType.DESCRIPTION);
    }

    @Override
    public @NotNull List<OriginSwapper.LineData.LineComponent> getTitle() {
        return OriginSwapper.LineData.makeLineFor("Large Appetite", OriginSwapper.LineData.LineComponent.LineType.TITLE);
    }

    @EventHandler
    public void onEntityExhaustion(EntityExhaustionEvent event) {
        AbilityRegister.runForAbility(event.getEntity(), getKey(), () -> event.setExhaustion(event.getExhaustion() * 1.6f));
    }
}

File: core/src/main/java/com/starshootercity/abilities/MoreExhaustion.kt
package com.starshootercity.abilities

import com.starshootercity.OriginSwapper.LineData.Companion.makeLineFor
import com.starshootercity.OriginSwapper.LineData.LineComponent
import net.kyori.adventure.key.Key
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.entity.EntityExhaustionEvent

class MoreExhaustion : VisibleAbility, Listener {
    override fun getKey(): Key {
        return Key.key("origins:more_exhaustion")
    }

    override fun getDescription(): MutableList<LineComponent?> {
        return makeLineFor(
            "You exhaust much quicker than others, thus requiring you to eat more.",
            LineComponent.LineType.DESCRIPTION
        )
    }

    override fun getTitle(): MutableList<LineComponent?> {
        return makeLineFor("Large Appetite", LineComponent.LineType.TITLE)
    }

    @EventHandler
    fun onEntityExhaustion(event: EntityExhaustionEvent) {
        runForAbility(event.getEntity()) { player ->
            event.exhaustion = event.exhaustion * 1.6f
        }
    }
}
